integrated_channels/integrated_channel/management/commands/__init__.py

```diff
@@ -59,6 +47,36 @@ def add_arguments(self, parser):
             choices=INTEGRATED_CHANNEL_CHOICES.keys(),
         )
 
+    def get_integrated_channels(self, options, **filter_kwargs):
+        """
+        Generates a list of active integrated channels, filtered from the given options.
+
+        Raises errors when invalid options are encountered.
+
+        See ``add_arguments`` for the accepted options.
+
+        filter_kwargs is passed as an additional set of parameters that jobs can use to restrict
+        the database query used to retrieve relevant integrated channels.
+        """
+        enterprise_customer = self._get_enterprise_customer(options.get('enterprise_customer'))
+        if enterprise_customer:
+            filter_kwargs['enterprise_customer'] = enterprise_customer
+
+        channel_classes = self._get_channel_classes(options.get('channel'))
+
+        # Loop through each channel class (optionally for a specific enterprise customer)
+        for channel_class in channel_classes:
+            # Use Active channels only
+            integrated_channels = channel_class.objects.filter(active=True)
+
+            # Filter down to the integrated channels that are strictly relevant
+            if filter_kwargs:
+                integrated_channels = integrated_channels.filter(**filter_kwargs)
+
+            # Gen the learner data to each integrated channel
```

Comment:
I didn't change the logic here either actually, just moved it up above the proclaimed private functions. I have no clue what this means and why that's here though, so I'll just scratch it, because the code looks readable enough itself.